EIPS/eip-2357.md

```diff
@@ -0,0 +1,37 @@
+---
+eip: 2357
+title: Total Difficulty in Block Header
+author: Tomasz K. Stanczak (@tkstanczak)
+discussions-to: https://ethereum-magicians.org/t/eip-2357-replace-difficulty-with-total-difficulty-in-block-headers/4233
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-04-29
+---
+
+## Simple Summary
+
+This document proposes to replace Dififculty with Total Difficulty in the Ethereum block headers..
+
+## Abstract
+
+Replace field Difficulty with TotalDifficulty (sum of difficulties since genesis) in the Ethereum block header.
+
+## Motivation
+
+Simplify total difficulty verification in stateless clients.
+
+## Specification
+
+Difficulty field will be replaced with TotalDifficulty (sum of difficulties since genesis) in the Ethereum block header.
```

Comment:
Is that really all what needs to be in the spec? (I'm not sure)

Reply:
I suspect the spec will need something like:
> on and after block FORK_BLOCK_NUMBER, the block header will include `totalDifficulty` which is a uint256 value that represents the sum of difficulties from of all blocks since genesis and `difficulty` field will no longer be present in the block header.

I'm just guessing though... I'm assuming by "the spec" you are referring to the Yellow Paper?

Reply:
@axic I understand that you point at implications in header validation and so on - I think this is worth discussing. Header validation requires parent anyway but the seal validation currently does not and it will after thic change. The implication is that invalid seal blocks may invoke one header lookup in the DB before the block is rebroadcasted but I think that all clients do a full header validation before broadcasting.

Reply:
The moment the parent is loaded the difficulty field can be calculated as the difference between parent and child. Worth reviewing if that works for AuRa chains but I think this will be ok. It does work fine for Clique.

Reply:
It may be worth finding out from @AlexeyAkhunov whether this change would work for TurboGeth.  The only reason I call it out specifically is because I know their DB structure is *significantly* different from other clients so anything that requires lookups of previous blocks isn't necessarily the same for them.  I don't *think* it matters in this case because their differences are primarily around the state DB, not the block headers...

Reply:
It is also a once-off look up at the transition block, otherwise it is just an increment on the previous `block.total_difficulty`.

Reply:
@axic there is an issue now that when a NewBlock message is sent over Eth62 it contains a TotalDifficulty field which is currently ignored by the clients because you cannot verify it, also when you exchange StatusMessage in Eth62 you mention total difficulty and then request headers and if someone is on a branch that is unknown to you you would need to ask for the entire branch headers to switch